test/e2e_node/services/namespace_controller.go

```diff
@@ -56,11 +59,15 @@ func (n *NamespaceController) Start() error {
 		return err
 	}
 	clientPool := dynamic.NewClientPool(config, registered.RESTMapper(), dynamic.LegacyAPIPathResolverFunc)
-	resources, err := client.Discovery().ServerPreferredNamespacedResources()
-	if err != nil {
-		return err
+	gvrFn := func() []k8s_unversioned.GroupVersionResource {
```

Comment:
put this into a library rather than duplicating?
(I am working on sharing code between kube ns controller and federation ns controller, so that is another place I will have to duplicate this)